src/plugins/decor/index.tsx

```diff
@@ -0,0 +1,195 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import ErrorBoundary from "@components/ErrorBoundary";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { findByCodeLazy } from "@webpack";
+import { Button, SelectedChannelStore, useEffect, UserStore } from "@webpack/common";
+
+import { CDN_URL, RAW_SKU_ID, SKU_ID } from "./lib/constants";
+import { useAuthorizationStore } from "./lib/stores/AuthorizationStore";
+import { useCurrentUserDecorationsStore } from "./lib/stores/CurrentUserDecorationsStore";
+import { useUsersDecorationsStore } from "./lib/stores/UsersDecorationsStore";
+import showAuthorizationModal from "./lib/utils/showAuthorizationModal";
+import { setDecorationGridDecoration, setDecorationGridItem } from "./ui/components";
+import { openChangeDecorationModal } from "./ui/modals/ChangeDecorationModal";
+
+const CustomizationSection = findByCodeLazy(".customizationSectionBackground");
+
+export default definePlugin({
+    name: "Decor",
+    description: "Create and use your own custom avatar decorations, or pick your favorite from the presets.",
+    authors: [Devs.FieryFlames],
+    patches: [
+        // Patch UserStore to include Decor avatar decorations when getting users
+        {
+            find: "getUserStoreVersion",
+            replacement: {
+                match: /(?<=getUser\(\i\){.{10,50}return )(\i\[\i\])/,
+                replace: "$self.patchUser($1)"
+            }
+        },
+        // Patch MediaResolver to return correct URL for Decor avatar decorations
+        {
+            find: "getAvatarDecorationURL:",
+            replacement: {
+                match: /(function \i\(\i\){)(let{avatarDecoration)/,
+                replace: "$1const vcDecorDecoration=$self.getDecorAvatarDecorationURL(arguments[0]);if(vcDecorDecoration)return vcDecorDecoration;$2"
+            }
+        },
+        // Patch profile customization settings to include Decor section
+        {
+            find: "DefaultCustomizationSections",
+            replacement: {
+                match: /{user:\i},"decoration"\),/,
+                replace: "$&$self.DecorSection(),"
+            }
+        },
+        // Decoration modal module
+        {
+            find: ".decorationGridItem",
+            replacement: [{
+                match: /(\i=)(\i=>{let{children:\i)/,
+                replace: "$1$self.DecorationGridItem=$2"
+            },
+            {
+                match: /(\i=)(\i=>{let{user:\i,avatarDecoration)/,
```

Comment:
you don't seem to be using this group other than immediately reinserting, so just make it a lookbehind. then you also don't need the other group
```suggestion
                match: /(?<=\i=)\i=>{let{user:\i,avatarDecoration/,
```